lib/repl.js

```diff
@@ -1229,8 +1229,8 @@ function complete(line, callback) {
       paths = ArrayPrototypeConcat(module.paths, CJSModule.globalPaths);
     }
 
-    for (let dir of paths) {
-      dir = path.resolve(dir, subdir);
+    for (let i = 0; i < paths.length; i++) {
+      const dir = path.resolve(paths[i], subdir);
```

Comment:
Other alternative:
```suggestion
    ArrayPrototypeForEach(paths, (dir) => {
      const dir = path.resolve(dir, subdir);
```

Reply:
Do u think I should change all such `for-loop`s to use `ArrayPrototypeForEach`? I was wondering why `ArrayPrototypeForEach` isn't used much in the rest of the codebase (performance perhaps?).

Reply:
> Do u think I should change all such for-loops to use ArrayPrototypeForEach?

I guess it's worth the try. I can launch a Benchmark CI after you change all loops to see if there is any perf difference.